### PR TITLE
[Refactor] Remove agent status

### PR DESCRIPTION
**Pull request by yiguolei**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

There are 3 error code types in BE: OLAPStatus AgentStatus Status. It is very confused and sometimes conflict during write code. I will try to unify them to Status.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by jackwener — COMMENTED** [review]
LGTM

**Comment by jackwener** [comment]
UT didn't pass. I can't ensure the correctness, It needs to build local and run all unit tests;

**Comment by morningman** [comment]
> Details

ut has been fixed. you need to rebase the master.

File: be/src/agent/pusher.cpp [hunk 1]
```diff
@@ -47,16 +47,15 @@ Pusher::Pusher(OLAPEngine* engine, const TPushReq& push_req)
 
 Pusher::~Pusher() {}
 
-AgentStatus Pusher::init() {
-    AgentStatus status = DORIS_SUCCESS;
+Status Pusher::init() {
+    Status status = Status::OK();
```
Comment (morningman) [inline be/src/agent/pusher.cpp:51]: Remove this line and define it at line 74

File: be/src/agent/pusher.cpp [hunk 2]
```diff
@@ -73,7 +72,7 @@ AgentStatus Pusher::init() {
     // Set download param
     string tmp_file_dir;
     status = _get_tmp_file_dir(olap_table->storage_root_path_name(), &tmp_file_dir);
-    if (status != DORIS_SUCCESS) {
+    if (status != Status::OK()) {
         LOG(WARNING) << "get local path failed. tmp file dir: " << tmp_file_dir;
```
Comment (morningman) [inline be/src/agent/pusher.cpp:76]: print `st.get_error_msg()`

File: be/src/agent/task_worker_pool.cpp [hunk 3]
```diff
@@ -1541,11 +1523,11 @@ void TaskWorkerPool::_move_dir_thread_callback() {
         TStatus task_status;
 
         // TODO: move dir
-        AgentStatus status =
+        Status status =
                 _move_dir(move_dir_req.tablet_id, move_dir_req.schema_hash, move_dir_req.src,
                           move_dir_req.job_id, true /* TODO */, &error_msgs);
 
-        if (status != DORIS_SUCCESS) {
+        if (status != Status::OK()) {
```
Comment (morningman) [inline be/src/agent/task_worker_pool.cpp:1530]: use `status.ok()`.
Better to modify all other places.

File: be/src/agent/task_worker_pool.cpp [hunk 4]
```diff
@@ -1581,7 +1563,7 @@ AgentStatus TaskWorkerPool::_move_dir(const TTabletId tablet_id, const TSchemaHa
         LOG(INFO) << "failed to get tablet. tablet_id:" << tablet_id
                   << ", schema hash:" << schema_hash;
         error_msgs->push_back("failed to get tablet");
-        return DORIS_TASK_REQUEST_ERROR;
+        return Status::InvalidArgument("Could not find tablet");
```
Comment (morningman) [inline be/src/agent/task_worker_pool.cpp:1566]: If using Status, we don't need `error_msgs` anymore.

File: be/src/agent/pusher.cpp (hunk 1 above)
Comment (yiguolei) [reply be/src/agent/pusher.cpp:51]: I already removed pusher.cpp file in next pr.

File: be/src/agent/pusher.cpp (hunk 2 above)
Comment (yiguolei) [reply be/src/agent/pusher.cpp:76]: I have already removed pusher.cpp file in next pr.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**